test/benchmarks.cpp

```diff
@@ -0,0 +1,460 @@
+//////////////////////////////////////////////////////////////////////////////////////
+//
+// (C) Daniel Strano 2017, 2018. All rights reserved.
+//
+// This is a multithreaded, universal quantum register simulation, allowing
+// (nonphysical) register cloning and direct measurement of probability and
+// phase, to leverage what advantages classical emulation of qubits can have.
+//
+// Licensed under the GNU General Public License V3.
+// See LICENSE.md in the project root or https://www.gnu.org/licenses/gpl-3.0.en.html
+// for details.
+
+#include <atomic>
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "catch.hpp"
+#include "qfactory.hpp"
+
+#include "tests.hpp"
+
+using namespace Qrack;
+
+#define EPSILON 0.001
+#define REQUIRE_FLOAT(A, B) do {                                            \
+        double __tmp_a = A;                                                 \
+        double __tmp_b = B;                                                 \
+        REQUIRE(__tmp_a < (__tmp_b + EPSILON));                             \
+        REQUIRE(__tmp_b > (__tmp_b - EPSILON));                             \
+    } while (0);
+
+void benchmarkLoop(std::function<void(QInterfacePtr, int)> fn) {
+
+    const int ITERATIONS = 100;
+
+    std::cout<<std::endl;
+    std::cout<<ITERATIONS<<" iterations";
+    std::cout<<std::endl;
+    std::cout<<"# of Qubits, ";
+    std::cout<<"Average Time (ms), ";
+    std::cout<<"Sample Std. Deviation (ms), ";
+    std::cout<<"Fastest (ms), ";
+    std::cout<<"1st Quartile (ms), ";
+    std::cout<<"Median (ms)";
+    std::cout<<"3rd Quartile (ms), ";
+    std::cout<<"Slowest (ms)"<<std::endl;
+
+    clock_t startClock, tClock, allClock;
+    clock_t trialClocks[ITERATIONS];
+
+    int i, numBits;
+
+    double avgt, stdet;
+
+    // Grover's search inverts the function of a black box subroutine.
+    // Our subroutine returns true only for an input of 100.
+    for (numBits = 3; numBits <= 20; numBits++) {
+        QInterfacePtr qftReg = CreateQuantumInterface(testEngineType, testSubEngineType, numBits, 0, rng);
+        startClock = clock();
+        allClock = startClock;
+        tClock = startClock;
```

Comment:
Unnecessary; remove.

Reply:
I don't see how this is unnecessary. The clocks have to be measured against the initial machine clock ticks.

Reply:
Last time I read this, it looked like the tClock assignment was redundant.